compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -588,62 +588,87 @@ OMR::ResolvedMethodSymbol::induceOSRAfter(TR::TreeTop *insertionPoint, TR_ByteCo
 
    if (self()->supportsInduceOSR(induceBCI, block, self()->comp()))
       {
-      TR::CFG *cfg = self()->comp()->getFlowGraph();
-      cfg->setStructure(NULL);
-      TR::TreeTop *remainderTree = insertionPoint->getNextTreeTop();
-      if (remainderTree->getNode()->getOpCodeValue() != TR::BBEnd)
+      self()->genInduceOSRAfter(insertionPoint, induceBCI, branch, extendRemainder, offset, lastTreeTop);
+      return true;
+      }
+
+   return false;
+   }
+
+bool
+OMR::ResolvedMethodSymbol::induceOSRAfter(TR::TreeTop *insertionPoint, TR::Node* osrNode, TR::TreeTop* branch,
```

Comment:
Where is this new API being used? inducing an OSR only conceptually needs and insertion point and a BCI so I'm not sure what this new API is going to get us? A more detailed discussion of this API and why it would be used would be helpful. Given an induction is something you expect to happen I'm not sure why the true false etc here is helpful...